src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -270,12 +270,24 @@ public static IHostBuilder SetAzureFunctionsConfigurationRoot(this IHostBuilder
 
         internal static void ConfigureApplicationInsights(HostBuilderContext context, ILoggingBuilder builder)
         {
-            string appInsightsKey = context.Configuration[EnvironmentSettingNames.AppInsightsInstrumentationKey];
+            string appInsightsInstrumentationKey = context.Configuration[EnvironmentSettingNames.AppInsightsInstrumentationKey];
+            string appInsightsConnectionString = context.Configuration[EnvironmentSettingNames.AppInsightsConnectionString];
 
             // Initializing AppInsights services during placeholder mode as well to avoid the cost of JITting these objects during specialization
-            if (!string.IsNullOrEmpty(appInsightsKey) || SystemEnvironment.Instance.IsPlaceholderModeEnabled())
+            if (!string.IsNullOrEmpty(appInsightsInstrumentationKey) || !string.IsNullOrEmpty(appInsightsConnectionString) || SystemEnvironment.Instance.IsPlaceholderModeEnabled())
             {
-                builder.AddApplicationInsightsWebJobs(o => o.InstrumentationKey = appInsightsKey);
+                builder.AddApplicationInsightsWebJobs(o =>
+                {
+                    if (!string.IsNullOrEmpty(appInsightsConnectionString))
+                    {
+                        o.ConnectionString = appInsightsConnectionString;
+                    }
+                    else
+                    {
+                        o.InstrumentationKey = appInsightsInstrumentationKey;
+                    }
```

Comment:
This is a bit of a nit -- but couldn't we just copy both into the options and not worry about what their values are? The options code handles if they're both set already. It keeps our logic here simpler.